src/sentry/monitors/logic/mark_failed.py

```diff
@@ -304,6 +326,41 @@ def create_issue_platform_occurrence(
     )
 
 
+HUMAN_FAILURE_STATUS_MAP: Mapping[int, str] = {
+    CheckInStatus.ERROR: "error",
+    CheckInStatus.MISSED: "missed",
+    CheckInStatus.TIMEOUT: "timeout",
+}
+
+
+def get_failure_reason(failed_checkins: Sequence[SimpleCheckIn]):
+    """
+    Builds a humam readible strong from a list of failed check-ins.
+
+    "3 missed check-ins detected"
+    "2 missed check-ins, 1 timeout check-in and 1 error check-in were detected"
+    "A failed check-in was detected"
+    """
+    status_counts = Counter(
+        HUMAN_FAILURE_STATUS_MAP[cast(int, checkin["status"])]
+        for checkin in failed_checkins
+        if checkin["status"] in HUMAN_FAILURE_STATUS_MAP.keys()
+    )
+
+    if sum(status_counts.values()) == 1:
+        return _("An %(status)s check-in was detected") % {"status": list(status_counts.keys())[0]}
```

Comment:
not familiar but will the string logic here correctly do `A` vs 'An`?

Reply:
good point. Let me see what django has for a e i o and u and sometimes y

Reply:
worth adding a test for that